File: app/src/main/java/in/smslite/viewHolder/CompleteSmsSentViewHolder.java
package in.smslite.viewHolder;

import android.preference.PreferenceManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import in.smslite.R;
import in.smslite.adapter.CompleteSmsAdapter;
import in.smslite.utils.TimeUtils;


public class CompleteSmsSentViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
  private static final String TAG = CompleteSmsSentViewHolder.class.getSimpleName();
  public static boolean tryFailedSms;
  private Long failedSmsTime;
  private View view;
  private TextView completeMsg, timeView;
  private LinearLayout linearLayout;
  private TextView smsStatusTextView;
  private String address;
  private CompleteSmsAdapter.SendTextSms sendTextSmsListener;
  public CompleteSmsSentViewHolder(View itemView, String address, CompleteSmsAdapter.SendTextSms sendTextSmsListener) {
    super(itemView);
    view = itemView;
    this.address = address;
    this.sendTextSmsListener = sendTextSmsListener;

    linearLayout = itemView.findViewById(R.id.card_sent_linear_layout_id);
    completeMsg = (TextView) view.findViewById(R.id.body);
    timeView = (TextView) view.findViewById(R.id.timestamp);
    smsStatusTextView = (TextView) view.findViewById(R.id.user_sms_status_id);
    smsStatusTextView.setOnClickListener(this);
  }

  public void setCompleteMsg(String body) {
    completeMsg.setText(body);
  }

  public void setTime(long x) {
    this.failedSmsTime = x;
    String timea = TimeUtils.getPrettyElapsedTime(x);
    timeView.setText(timea);
  }

  @Override
  public void onClick(View v) {
    Log.d(TAG, "trySendFailedSms");
    Log.d(TAG, Long.toString(failedSmsTime));
    tryFailedSms = true;
    int category = PreferenceManager.getDefaultSharedPreferences(v.getContext()).getInt(view.getContext().getString(R.string.dialog_option),0);
    sendTextSmsListener.onSendTextSms( failedSmsTime, address, category);
  }

  public void setItemBackgroundBlack() {
    linearLayout.setBackgroundResource((R.drawable.sms_sent_background_black));
  }

  public void setItemBackgroundBlue() {
    linearLayout.setBackgroundResource(R.drawable.sms_sent_background);
  }

  public void setBackgroundColorWhite() {
    view.setBackgroundColor(view.getContext().getResources().getColor(R.color.white_pure));
  }

  public void setBackgroundColor() {
    view.setBackgroundColor(view.getContext().getResources().getColor(R.color.item_selected));
  }

}

File: app/src/main/java/in/smslite/viewHolder/CompleteSmsSentViewHolder.kt
package `in`.smslite.viewHolder

import `in`.smslite.R
import `in`.smslite.adapter.CompleteSmsAdapter
import `in`.smslite.utils.TimeUtils
import android.preference.PreferenceManager
import android.util.Log
import android.view.View
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class CompleteSmsSentViewHolder(private val view: View, private val address: String, sendTextSmsListener: CompleteSmsAdapter.SendTextSms) : RecyclerView.ViewHolder(view), View.OnClickListener {
    private var failedSmsTime: Long? = null
    private val completeMsg: TextView
    private val timeView: TextView
    private val linearLayout: LinearLayout
    private val smsStatusTextView: TextView
    private val sendTextSmsListener: CompleteSmsAdapter.SendTextSms
    fun setCompleteMsg(body: String?) {
        completeMsg.text = body
    }

    fun setTime(x: Long) {
        failedSmsTime = x
        val timea = TimeUtils.getPrettyElapsedTime(x)
        timeView.text = timea
    }

    override fun onClick(v: View) {
        Log.d(TAG, "trySendFailedSms")
        Log.d(TAG, java.lang.Long.toString(failedSmsTime!!))
        tryFailedSms = true
        val category = PreferenceManager.getDefaultSharedPreferences(v.context).getInt(view.context.getString(R.string.dialog_option), 0)
        sendTextSmsListener.onSendTextSms(failedSmsTime, address, category)
    }

    fun setItemBackgroundBlack() {
        linearLayout.setBackgroundResource(R.drawable.sms_sent_background_black)
    }

    fun setItemBackgroundBlue() {
        linearLayout.setBackgroundResource(R.drawable.sms_sent_background)
    }

    fun setBackgroundColorWhite() {
        view.setBackgroundColor(view.context.resources.getColor(R.color.white_pure))
    }

    fun setBackgroundColor() {
        view.setBackgroundColor(view.context.resources.getColor(R.color.item_selected))
    }

    companion object {
        private val TAG = CompleteSmsSentViewHolder::class.java.simpleName
        var tryFailedSms = false
    }

    init {
        this.sendTextSmsListener = sendTextSmsListener
        linearLayout = itemView.findViewById(R.id.card_sent_linear_layout_id)
        completeMsg = view.findViewById<View>(R.id.body) as TextView
        timeView = view.findViewById<View>(R.id.timestamp) as TextView
        smsStatusTextView = view.findViewById<View>(R.id.user_sms_status_id) as TextView
        smsStatusTextView.setOnClickListener(this)
    }
}